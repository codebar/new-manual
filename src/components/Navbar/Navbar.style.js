import styled from 'styled-components';
import { Link } from 'react-router-dom';
import {
  primaryBlueColor,
  primaryPinkColor,
  primaryGreenColor,
  primaryDarkgreyColor,
} from '../../globalstyle/GlobalStyle';
import { COLORS } from '../../globalstyle/colors';

const Logo = styled.img``;

const textDecoration = 'text-decoration: none;';

const ExternalLink = styled.a`
  ${textDecoration}
  color: #4bafff;
`;

const DonateButton = styled.a`
  ${textDecoration}
  background-color: #4bafff;
  color: white;
  border-radius: 50px;
  padding: 0.75rem;
  max-width: 6rem;
  text-align: center;
`;

const MenuLink = styled(Link)`
  ${textDecoration}
  color: black;
`;

const Nav = styled.nav`
  display: grid;
  grid-template-columns: 3fr 1fr 1fr 1fr 1fr 1fr;
  gap: 1px 1px;
  grid-template-areas: 'Logo Logo Logo about sponsors organisers donate codebar';
  margin: 1.5rem 1rem;
  align-items: center;
  font-size: 1.15rem;
  color: ${COLORS.green};
`;

export { Logo, Nav, MenuLink, ExternalLink, DonateButton };
