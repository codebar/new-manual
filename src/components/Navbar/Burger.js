import React, { useState } from 'react';
import { StyledBurger } from './Navbar.style';
import RightNav from './RightNav';

const Burger = () => {
    const [open, setOpen] = useState(false);

    return (
        <>
            <StyledBurger open={open} onClick={() => setOpen(!open)}>
            <div />
            <div />
            <div />
            </StyledBurger>
            <RightNav open={open}/>
        </>
    )
}

export default Burger;