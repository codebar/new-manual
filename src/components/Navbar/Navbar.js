import React from 'react';
import logo from '../../assets/image/logo.png';
import { Nav, Logo, MenuLink, ExternalLink, DonateLink } from './Navbar.style';

const Navbar = () => {
  return (
    <Nav>
      <MenuLink to="/" className="Logo">
        <Logo src={logo} alt="codebar-logo" />
      </MenuLink>
      <MenuLink to="/about" className="About">
        About
      </MenuLink>
      <MenuLink to="/sponsors" className="Sponsors">
        Sponsors
      </MenuLink>
      <MenuLink to="/organisers" className="Organisers">
        Organisers
      </MenuLink>

      <ExternalLink
        href="https://codebar.io/"
        className="Codebar"
        target="_blank"
      >
        codebar.io
      </ExternalLink>

      <DonateLink
        href="https://codebar.enthuse.com/donate/#!/"
        className="Donate"
        target="_blank"
      >
        Donate
      </DonateLink>
    </Nav>
  );
};

export default Navbar;
