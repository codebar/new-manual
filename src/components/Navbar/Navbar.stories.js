import React from 'react';
import Navbar from './Navbar';
import { MemoryRouter } from 'react-router';

export default { title: 'Navbar', component: Navbar };

export const nav = () => <MemoryRouter>{Navbar()}</MemoryRouter>;
