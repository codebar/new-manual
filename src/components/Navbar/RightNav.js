import React from 'react';
import {DonateLink, ExternalLink, MenuLink, NavWrapper } from './Navbar.style'

const RightNav = ({ open }) => {
    return (
        <NavWrapper open={open}>
            <li>
            <MenuLink to="/about" className="About">
            About
            </MenuLink>
            </li>
            <li>
            <MenuLink to="/sponsors" className="Sponsors">
            Sponsors
             </MenuLink>
            </li>
            <li>
            <MenuLink to="/organisers" className="Organisers">
            Organisers
             </MenuLink>
            </li>
            <li>
            <ExternalLink
            href="https://codebar.io/"
            className="Codebar"
            target="_blank"
        >
            codebar.io
        </ExternalLink>
            </li>

            <li>
            <DonateLink
            href="https://codebar.enthuse.com/donate/#!/"
            className="Donate"
            target="_blank"
        >
            Donate
        </DonateLink>
            </li>
        </NavWrapper>
    )
}


export default RightNav;