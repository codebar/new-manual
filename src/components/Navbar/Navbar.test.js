import React from 'react';
import Navbar from './Navbar';

import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router';

it('matches snapshot', () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <Navbar />
      </MemoryRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
