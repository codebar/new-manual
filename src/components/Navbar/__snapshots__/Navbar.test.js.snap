// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<nav
  className="sc-AxhUy bYjeWk"
>
  <a
    className="sc-AxhCb kITrgR logoContainer"
    href="/"
    onClick={[Function]}
  >
    <img
      alt="codebar-logo"
      className="sc-AxjAm StDqN logo"
      src="website-logo.png"
    />
  </a>
  <a
    className="sc-AxhCb kITrgR About"
    href="/about"
    onClick={[Function]}
  >
    About
  </a>
  <a
    className="sc-AxhCb kITrgR Sponsors"
    href="/sponsors"
    onClick={[Function]}
  >
    Sponsors
  </a>
  <a
    className="sc-AxhCb kITrgR Organisers"
    href="/organisers"
    onClick={[Function]}
  >
    Organisers
  </a>
  <a
    className="sc-AxirZ hYzhrQ Codebar"
    href="https://codebar.io/"
    target="_blank"
  >
    codebar.io
  </a>
  <a
    className="sc-AxiKw hncjQH Donate"
    href="https://codebar.enthuse.com/donate/#!/"
    target="_blank"
  >
    Donate
  </a>
</nav>
`;
